Sources/BraintreeCoreSwift/BTJSON.swift

```diff
@@ -0,0 +1,241 @@
+/// ```
+@objcMembers public class BTJSON: NSObject {
+    var value: Any? = [:]
+
+    // MARK: Initializers
+
+    public override init() { }
+
+    ///  Initialize with a value.
+    /// - Parameter value: The value to initialize with.
+    public convenience init(value: Any?) {
+        self.init()
+        self.value = value
+    }
+
+    /// Initialize with data.
+    /// - Parameter data: The `Data` to initialize with.
+    public convenience init(data: Data) {
+        do {
+            let value = try JSONSerialization.jsonObject(with: data, options: .allowFragments)
+            self.init(value: value)
+        } catch {
+            self.init(value: error)
+        }
+    }
+
+    // MARK: JSON Type Checks
+
+    /// Checks if the `BTJSON` is a `String`
+    /// - Returns: `true` if this instance of `BTJSON` is a valid `String`
+    public var isString: Bool {
+        value is String
+    }
+
+    /// Checks if the `BTJSON` is a `Bool`
+    /// - Returns: `true` if this instance of `BTJSON` is a valid `Bool`
+    public var isBool: Bool {
+        value is Bool
+    }
+
+    /// Checks if the `BTJSON` is a `NSNumber`
+    /// - Returns: `true` if this instance of `BTJSON` is a valid `NSNumber`
+    public var isNumber: Bool {
+        value is NSNumber
+    }
+
+    /// Checks if the `BTJSON` is a `[Any]`
+    /// - Returns: `true` if this instance of `BTJSON` is a valid `[Any]`
+    public var isArray: Bool {
+        value is [Any]
+    }
+
+    /// Checks if the `BTJSON` is a `[String: Any]`
+    /// - Returns: `true` if this instance of `BTJSON` is a valid `[String: Any]`
+    public var isObject: Bool {
+        value is [String: Any]
+    }
+
+    /// Checks if the `BTJSON` is an error.
+    /// - Returns: `true` if this instance of `BTJSON` is not valid.
+    public var isError: Bool {
+        value is NSError
+    }
+
+    /// Checks if the `BTJSON` is a value representing `true`
+    /// - Returns: `true` if this instance of `BTJSON` is `true`
+    public var isTrue: Bool {
+        value as? Bool == true
+    }
+
+    /// Checks if the `BTJSON` is a value representing `false`
+    /// - Returns: `true` if this instance of `BTJSON` is `false`
+    public var isFalse: Bool {
+        value as? Bool == false
+    }
+
+    /// Checks if the `BTJSON` is a value representing `nil`
+    /// - Returns: `true` if this instance of `BTJSON` is `nil`
+    public var isNull: Bool {
+        value is NSNull
+    }
+
+    // MARK: Subscripting
+
+    ///  Indexes into the JSON as if the current value is an object
+    ///
+    /// Notably, this method will always return successfully; however, if the value is not an object, the JSON will wrap an error.
+    public subscript(index: Int) -> BTJSON {
+        if value is NSError {
+            return self
+        }
+
+        guard let value = value as? [Any],
+                index < value.count else {
```

Comment:
```suggestion
        guard let value = value as? [Any], index < value.count else {
```
🧶 : we can flatten this out to improve readability.